modules/javafx.graphics/src/main/java/com/sun/glass/ui/GlassRobot.java

```diff
@@ -0,0 +1,363 @@
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.sun.glass.ui;
+
+import java.lang.annotation.Native;
+import java.nio.ByteBuffer;
+import java.nio.IntBuffer;
+import java.util.Objects;
+
+import javafx.scene.image.Image;
+import javafx.scene.image.PixelWriter;
+import javafx.scene.image.WritableImage;
+import javafx.scene.input.KeyCode;
+import javafx.scene.input.MouseButton;
+import javafx.scene.paint.Color;
+import javafx.stage.Screen;
+
+import com.sun.javafx.image.PixelUtils;
+
+public abstract class GlassRobot {
+
+    @Native public static final int MOUSE_LEFT_BTN    = 1 << 0;
+    @Native public static final int MOUSE_RIGHT_BTN   = 1 << 1;
+    @Native public static final int MOUSE_MIDDLE_BTN  = 1 << 2;
+
+    /**
+     * Initializes any state necessary for this {@code Robot}. Called by
+     * the {@code Robot} constructor.
+     */
+    public abstract void create();
+
+    /**
+     * Frees any resources allocated by this {@code Robot}.
+     */
+    public abstract void destroy();
+
+    /**
+     * Presses the specified {@link KeyCode} key.
+     *
+     * @param keyCode the key to press
+     */
+    public abstract void keyPress(KeyCode keyCode);
+
+    /**
+     * Releases the specified {@link KeyCode} key.
+     *
+     * @param keyCode the key to release
+     */
+    public abstract void keyRelease(KeyCode keyCode);
+
+    /**
+     * Returns the current mouse x-position.
+     *
+     * @return the current mouse x-position
+     */
+    public abstract double getMouseX();
+
+    /**
+     * Returns the current mouse y-position.
+     *
+     * @return the current mouse y-position
+     */
+    public abstract double getMouseY();
+
+    /**
+     * Moves the mouse to the specified (x,y) screen coordinates.
+     *
+     * @param x screen coordinate x to move the mouse to
+     * @param y screen coordinate y to move the mouse to
+     */
+    public abstract void mouseMove(double x, double y);
+
+    /**
+     * Presses the specified {@link MouseButton}s.
+     *
+     * @param buttons the mouse buttons to press
+     */
+    public abstract void mousePress(MouseButton... buttons);
+
+    /**
+     * Releases the specified {@link MouseButton}s.
+     *
+     * @param buttons the mouse buttons to release
+     */
+    public abstract void mouseRelease(MouseButton... buttons);
+
+    /**
+     * Scrolls the mouse wheel by the specified amount. Positive {@code wheelAmt}s
+     * scroll up whereas negative scroll down.
+     *
+     * @param wheelAmt the (signed) amount to scroll the wheel
+     */
+    public abstract void mouseWheel(int wheelAmt);
+
+    /**
+     * Returns the {@link Color} of the pixel at the specified screen coordinates
+     * of the primary screen.
```

Comment:
I did not check the implementation of this method, but can you, please, confirm that this method will return the real "pixel" color at point x,y on the HiDPI screen. I have asked because on the HiDPI screen the unit at point (x,y) may have 4 pixels, and it is unclear, which color will be returned:
 - The color which interpolate 4 pixels
 - The color of the top/left pixel.